### ISSUE TITLE
[VOTE for this feature if interested] - Add a way to change the preset temperature from VTherm UI card

### ISSUE THREAD

**Issue by gunmalmg** [jmcollin78/versatile_thermostat#186]
Thank you for a greathe thermostat that gives us consistant presets for all our different underlaying climate entities and now with Self-regulation over AC-Heatpumps. 👏

But I have for a long time wished that the preset-temperatures could be accessible as Configation_entries under the VT-device. That would be a much more convenient way to edit the set points true the Lovelace 

Hope this can be achieved 

Regards 
Gunnar M

**Comment by jmcollin78** [comment]
Hello @gunmalmg ,

All the configuration is accessible through config_entries which is the standard mecanism in HA to persist some configuration.

If you look at the homeassistant/.storage/core.config_entries you will find all your VTherm confitguration:

```
{
        "entry_id": "dcb48e2a03dbc60b562ccafae0bc2bb6",
        "version": 1,
        "domain": "versatile_thermostat",
        "title": "Thermostat musique",
        "data": {
          "name": "Thermostat musique",
          "heater_entity_id": "switch.radiateur_musique",
          "temperature_sensor_entity_id": "sensor.temperature_musique",
          "cycle_min": 7,
          "proportional_function": "tpi",
          "proportional_bias": 0.25,
          "eco_temp": 15.0,
          "away_temp": 15.0,
          "boost_temp": 20.0,
          "comfort_temp": 19.0,
          "power_temp": 12.0,
          "window_delay": 30,
          "motion_delay": 30,
          "motion_preset": "comfort",
          "no_motion_preset": "eco",
          "power_sensor_entity_id": "sensor.total_puissance_consommee_net_instantanee"
...
```

I'm not sure to understand your need. What do you want to do with this configuration entries ?

**Comment by gunmalmg** [comment]
Hallo Jmcollin

Maby I´m not using the right terms. But my point is that I want it to be easy to alter the temp-set points via the Lovelace. Without the need to make many Helpers and automations via service-calls. 
Other devices have many of there configuration settings exposed via the Device stings page. See attached screenshot example. That gives us the easy ability to just expose those config-setting via Lovelace to bee altered - Just like that. 
Very convenient and straight forward to implement. 

Hope this was clarifying :-) 

Regards Gunnar 


<img width="529" alt="Skjermbilde 2023-11-11 kl  22 17 06" src="https://github.com/jmcollin78/versatile_thermostat/assets/128496535/95d9077d-913f-4694-baa6-d53444e81386">

**Comment by jmcollin78** [comment]
Ah ok. I got it now. But you known that you can already change all the parameters with the Lovelace UI ?
Just go to Parameters / Integration / Click on the Versatile Thermostat integration and then on Configure at the right of the VTherm you want to configure. It is no more complicated than creating entities for configuration like you propose.

Or is there any issue with the configuration page ?

![Capture d’écran 2023-11-12 à 00 12 16](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/ea95ebad-ec0d-47f0-97a5-3c5530ed51b5)
![Capture d’écran 2023-11-12 à 00 12 42](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/f962473d-9abc-48c0-a48d-6b9a245d3a3b)

**Comment by gunmalmg** [comment]
I am fully aware of all this. It is how I configure the thermostat.

What I am seeking is the ability to let my wife or other normal users alter preset-points without being a computer nerd. I can't  let them into the configuration of the whole VT-Thermostat. They would certainly ruin the whole setup 😜😬
In the daily use of my HA-systems I need to let the ordinary user(not me as an administrator) - like my dad and others - be able to alter the preset-points in a secure and user-friendly way. 

So please provide us with the ability to add this settings in a convenient way to a Lovelace dashboard. (like the example screenshot) 

Hope this clarifies my request. 🙏

😀

Regards 
Gunnar


![Bilde](https://github.com/jmcollin78/versatile_thermostat/assets/128496535/2e505937-9458-4346-b030-539d65cd7acf)

**Comment by jmcollin78** [comment]
Yes the request is now more clear: you want an non admin user to change the target temperature.

Are you aware of this new card which enables you to change the target temperature in a very userfriendly way: https://github.com/jmcollin78/versatile-thermostat-ui-card

Because it is new, maybe you miss it.

In the HA philosophie, all configuration is done through the ConfigFlow and not directly on the device page. Device page displays the entities of the devices and not its configuration. So I guess Reolink does something special (ie create specific entities only for configuration). For me it is really not the standard in HA and I guess they don't have a configuration through the ConfigFlow (the UI of the integration).

I fear that both ways of configuration will overlap each other.

So for now, I'm not convience I will do that. But there is maybe a solution through the versatile-thermostat-ui-card. I will think about that.

**Comment by gunmalmg** [comment]
I am aware of the new card, but that card don't let the user alter the preset set-points, only the current(manual) setpoint. 

If you could let the ordinary users alter the preset-setpoints through the versatile-thermostat-ui-card it would bee a great solution 👏😎.  Then the goal is achieved 👍. 

Good path to a good solution 😀

Regards Gunnar

**Comment by jmcollin78** [comment]
Yes but have in mind that the modification is temporary. At each restart of the VTherm it will restore the value in the configuration which is stored by HA.
Like the service call does.

I cannot do differently with this solution.

**Comment by gunmalmg** [comment]
Do you mean that HA is not able to preserve settings over a restart ? That sound strange..
If so - and the only way to preserve altered presets is by altering your underlaying config - I will hope that you are able to alter your config in the background so it survives a restart. 
Am I on to sorting ?

**Comment by jmcollin78** [comment]
As far than I know, configuration of the integration (Parameters / Integration / configure) are not in the state and can be change only during the configuration phase. Those changes are persistent into a file (.storage/core.config_entry). For me this cannot be change during execution phase but only in configuration phase but I can be wrong.

Changing entities values (like in your screenshot) are of course persistent because they are part of the state of an entity. In your screenshot the integration uses entities state as configuration.

That make a big difference.

**Comment by jmcollin78** [comment]
This is a big change and I guess once you will have the good value, you won't change it anymore. My values are fix and I don't touch it since 2 years now.
But in experiment phase, I agree this will be a great enhancement.



# To community, please add a +1 in this post, if you are interested for this feature. (I will update the title)

**Comment by gunmalmg** [comment]
Thank you for your positive responce. 😀

This would be exiting - and a permanent enhancement that will really make your VT stand out compared to other integrations. 😎

Regards Gunnar

The Presets on the right side can perhaps follow the state of the AC. 
In Havc_mode=Cool it changes the Cool presets
In Havc_mode=Heat it changes the Heat presets
In Away_mode  it changes the Away presets, And the same for Home-Mode

**Comment by gunmalmg** [comment]
Maby you removes the "wontfix" label on the post :-)

**Comment by gunmalmg** [comment]
By the way - In Norway we need to tweak this presets during the year. Because we live in a climate that has grate variation in temperature during the year.  The presets for the spring will not function well in the winter, and the presets for summer will not function well in the autumn...and so on.

So this is highly relevant during the year - also after the initial experimental phase. For us up north 😀

**Comment by jfchadeyron** [comment]
I vote againt. Too many informations on this card.

**Comment by jmcollin78** [comment]
> I vote againt. Too many informations on this card. I use vertical cards for many thermostats. On smartphone, it is smaller. ![Capture d’écran 2023-11-17 à 11 28 16](https://user-images.githubusercontent.com/78847492/283763776-fbc67303-a477-4b05-be07-f65f3c7bdddc.png)

If we add more informations, they are toggable. Yo ucan always choose to not display it.

**Comment by gunmalmg** [comment]
It is not important for me that this is solved through the versatile-thermostat-ui-card. That was just a suggestion.
And I agree that this could clutter the card. But as you says - the extra functionality in the UI-card can be toggable. 
The core request/need is that we can alter the preset-temps through the UI and not only via the VT-Config. 
I think the best way would be to let this temp-presets be alterable in the UI by exposing them as config entities under the VT-Device config screen. 
Like the Floorheating thermostats I use. (see screenshot)
They expose some of there config-settings as config entities under the Thermostat Device. This gives us a easy way to integrate these config settings into the UI the same as any other entity in HA. And we can decide witch settings we find relevant to expose to the user.  

This seems to be a fairly common way for many integrations to let the user alter the config settings directly in the Device setting screen, or let us use the "Add to dashboard" button to add it to a UI-Dashboard.

Hope you find time and interest in working on this since we  In the North have to tweak/alter this presets during the changing seasons, and not just in a experimental phase. 😀

Regards Gunnar

![IMG_3378](https://github.com/jmcollin78/versatile_thermostat/assets/128496535/da77424e-f0b4-469f-b3cb-31085022c7a6)

**Comment by nicocolt** [comment]
Hello, It will be helpful if this feature will implemented. I manage my radiator only with preset mode and it is not working for instace with the VTherm UI card.

Best,
Nicolas

**Comment by jmcollin78** [comment]
Included into 4.3.0 and 0.5.0 Card Ui.

**Comment by mycanaletto** [comment]
Another possibility, perhaps simpler, would be to enter an `input_number` instead of the setpoint temperature. This way, you don't add entities, and you can use external setpoints...
